### PR TITLE
Bump tektoncd/pipeline to v0.12.1

### PR DESCRIPTION
**Pull request by piyush-garg**
This will bump tektoncd/pipeline to v0.12.1 for both
v1alpha and v1beta1 resources in v1beta1 client.
Also use the respective dep as in the upstream project

### CODE REVIEW

**Comment by rohanKanojia** [comment]
ok to test

**Comment by rohanKanojia** [comment]
Welcome back :smile:

**Comment by rohanKanojia** [comment]
@Fabian-K : Could you please review this whenever you find some time?

**Comment by Fabian-K** [comment]
Sure, unfortunately I might need 2-3 days to find some time for it :/

File: extensions/tekton/tests/src/test/java/io/fabric8/tekton/test/crud/V1alpha1PipelineCrudTest.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 import io.fabric8.tekton.pipeline.v1alpha1.Pipeline;
 import io.fabric8.tekton.pipeline.v1alpha1.PipelineBuilder;
 import io.fabric8.tekton.pipeline.v1alpha1.PipelineList;
-import io.fabric8.tekton.v1alpha1.internal.pipeline.pkg.apis.pipeline.v1alpha2.Param;
+import io.fabric8.tekton.v1alpha1.internal.pipeline.pkg.apis.pipeline.v1beta1.Param;
```
Comment (manusa) [inline extensions/tekton/tests/src/test/java/io/fabric8/tekton/test/crud/V1alpha1PipelineCrudTest.java:23]: The test suite is named `V1alpha1PipelineCrudTest`

Maybe we should create a `V1beta1PipelineCrudTest` suite with v1beta1 specific tests.

**Comment by rohanKanojia** [comment]
No rush, take your time

File: extensions/tekton/tests/src/test/java/io/fabric8/tekton/test/crud/V1alpha1PipelineCrudTest.java (hunk 1 above)
Comment (manusa) [reply extensions/tekton/tests/src/test/java/io/fabric8/tekton/test/crud/V1alpha1PipelineCrudTest.java:23]: Ok, sorry, missed the first part of the import. I suppose something has changed in the base v1alpha1 api (but this change is weird for me nonetheless)

File: extensions/tekton/generator-v1alpha1/go.mod [hunk 2]
```diff
@@ -2,10 +2,17 @@ module github.com/fabric8io/kubernetes-client/extensions/tekton/generator
 
 require (
 	github.com/fabric8io/kubernetes-client/generator v0.0.0
-	github.com/tektoncd/pipeline v0.10.2
```
Comment (Fabian-K) [inline extensions/tekton/generator-v1alpha1/go.mod:5]: as mentioned, v1alpha1 should not be updated to the latest version!

File: extensions/tekton/generator-v1beta1/go.mod [hunk 3]
```diff
@@ -2,10 +2,17 @@ module github.com/fabric8io/kubernetes-client/extensions/tekton/generator
...
+	k8s.io/apimachinery v0.17.3
+	k8s.io/client-go v0.17.3 // indirect
 )
 
-replace github.com/fabric8io/kubernetes-client/generator v0.0.0 => ./../../../generator
+replace (
+	github.com/fabric8io/kubernetes-client/generator v0.0.0 => ./../../../generator
+	k8s.io/api => k8s.io/api v0.16.5
+	k8s.io/apimachinery => k8s.io/apimachinery v0.16.5
+	k8s.io/client-go => k8s.io/client-go v0.16.5
```
Comment (Fabian-K) [inline extensions/tekton/generator-v1beta1/go.mod:15]: Do you know why this is necessary?

File: extensions/tekton/tests/src/test/java/io/fabric8/tekton/test/crud/V1alpha1PipelineCrudTest.java (hunk 1 above)
Comment (Fabian-K) [reply extensions/tekton/tests/src/test/java/io/fabric8/tekton/test/crud/V1alpha1PipelineCrudTest.java:23]: that's caused by the update of the v1alpha1 dependency

File: extensions/tekton/generator-v1beta1/go.mod (hunk 3 above)
Comment (piyush-garg) [reply extensions/tekton/generator-v1beta1/go.mod:15]: I did this to match with whatever is used in upstream https://github.com/tektoncd/pipeline/blob/release-v0.12.x/go.mod#L63

File: extensions/tekton/generator-v1alpha1/go.mod (hunk 2 above)
Comment (piyush-garg) [reply extensions/tekton/generator-v1alpha1/go.mod:5]: Yup I misunderstood it. I thought the v1beta1 client for only v1beta1 resources. I think it has both latest v1alpha1 and v1beta1. While v1alpha client is for old v1alpha1 resources only. In that case, we don't need to update v1alpha1 client then

**Comment by rohanKanojia** [comment]
@Fabian-K : Could you please review the update whenever you get time? Once you approve we can proceed to merge this.

File: extensions/tekton/generator-v1alpha1/go.sum [hunk 4]
```diff
@@ -6,14 +6,17 @@ cloud.google.com/go v0.44.1/go.mod h1:iSa0KzasP4Uvy3f1mN/7PiObzGgflwredwwASm/v6A
 cloud.google.com/go v0.44.2/go.mod h1:60680Gw3Yr4ikxnPRS/oxxkBccT6SA1yMk63TGekxKY=
 cloud.google.com/go v0.45.1/go.mod h1:RpBamKRgapWJb87xiFSdk4g1CME7QZg3uwTez+TSTjc=
 cloud.google.com/go v0.46.3/go.mod h1:a6bKKbmY7er1mI7TEI4lsAkts/mkhTSZK8w33B4RAg0=
+cloud.google.com/go v0.47.0 h1:1JUtpcY9E7+eTospEwWS2QXP3DEn7poB3E2j0jN74mM=
```
Comment (Fabian-K) [inline extensions/tekton/generator-v1alpha1/go.sum:9]: I´m not sure why there are changes in the go.sum for the v1alpha1. Anyway, they don´t seem to affect the resulting json. Therfore I´m fine with both options: reverting or keeping them

**Comment by rohanKanojia** [comment]
Thanks a lot @piyush-garg :heart:

**Comment by rohanKanojia** [comment]
[merge]

**Comment by rohanKanojia** [comment]
[merge]

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**